elasticsearch/examples/config/values.yaml

```diff
@@ -0,0 +1,38 @@
+---
+
+clusterName: "config"
+replicas: 1
+
+extraEnvs:
+  - name: ELASTIC_PASSWORD
+    valueFrom:
+      secretKeyRef:
+        name: elastic-credentials
+        key: password
+  - name: ELASTIC_USERNAME
+    valueFrom:
+      secretKeyRef:
+        name: elastic-credentials
+        key: username
+  - name: SLACK_URL
+    valueFrom:
+      secretKeyRef:
+        name: elastic-config-slack
+        key: slack_url
+
+esConfig:
+  elasticsearch.yml: |
+    path.data: /usr/share/elasticsearch/data
+
+secretMounts:
+  - name: elastic-config-secret
+    secretName: elastic-config-secret
+    path: /usr/share/elasticsearch/config/secret
+
+keystore:
+  enabled: true
+  strings:
+    bootstrap.password: '${ELASTIC_PASSWORD}'
+    xpack.notification.slack.account.monitoring.secure_url: '${SLACK_URL}'
+  files:
```

Comment:
cloud-on-k8s doesn't have a specific concept of file vs string keystore entries: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-es-secure-settings.html

I don't recall why that is, but I wonder if we can have the same concepts for this on both the chart and the operator?

Reply:
I remember a change was made (initiated by the cloud-on-k8s team) to have the Elasticsearch keystore treat all strings and files the same internally. 

My original assumption was that the operator would be doing something special to take these secrets, generate a keystore and create that as a secret. But their approach is also [fully init container based](https://github.com/elastic/cloud-on-k8s/blob/0f5aab786ea99f1374dceca1e86fca2c385de117/operators/pkg/controller/common/keystore/initcontainer.go#L33-L51). All secrets are mounted as files and then the init script loops over them and adds them to the keystore. 

I like this a lot more than my solution, mainly because you don't need to think about mapping the files and strings into the container yourself. And the other huge security advantage is that these secrets won't still exist as environment variables or mounts in the actual running Elasticsearch container. 

One downside to this approach is that it looks like the spec only allows a single secret. The downside of this being that you now need to fully recreate your keystore secret everytime you want to add something new. 

Making it a list would solve this so I'll ping the team and see what their thoughts on it are. 
```
spec:
  secureSettings:
    - secretName: your-secure-settings-secret
    - secretName: other-secure-settings-secret
```

I'm going to take a stab at implementing this in the same way.